src/DispatchListener.php

```diff
@@ -92,12 +93,14 @@ public function onDispatch(MvcEvent $e)
         try {
             $return = $controller->dispatch($request, $response);
         } catch (\Exception $ex) {
-            $e->setError($application::ERROR_EXCEPTION)
-                  ->setController($controllerName)
-                  ->setControllerClass(get_class($controller))
-                  ->setParam('exception', $ex);
-            $results = $events->trigger(MvcEvent::EVENT_DISPATCH_ERROR, $e);
-            $return = $results->last();
+            $e->setName(MvcEvent::EVENT_DISPATCH_ERROR);
+            $e->setError($application::ERROR_EXCEPTION);
+            $e->setController($controllerName);
+            $e->setControllerClass(get_class($controller));
```

Comment:
That's out of scope for this particular refactor.
On Oct 1, 2015 6:34 AM, "Marco Pivetta" <notifications@github.com> wrote:

> In src/DispatchListener.php
> <https://github.com/zendframework/zend-mvc/pull/31#discussion_r40903363>:
>
> > @@ -92,12 +93,14 @@ public function onDispatch(MvcEvent $e)
> >          try {
> >              $return = $controller->dispatch($request, $response);
> >          } catch (\Exception $ex) {
> > -            $e->setError($application::ERROR_EXCEPTION)
> > -                  ->setController($controllerName)
> > -                  ->setControllerClass(get_class($controller))
> > -                  ->setParam('exception', $ex);
> > -            $results = $events->trigger(MvcEvent::EVENT_DISPATCH_ERROR, $e);
> > -            $return = $results->last();
> > +            $e->setName(MvcEvent::EVENT_DISPATCH_ERROR);
> > +            $e->setError($application::ERROR_EXCEPTION);
> > +            $e->setController($controllerName);
> > +            $e->setControllerClass(get_class($controller));
>
> Can this be part of the event class instead?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-mvc/pull/31/files#r40903363>.
>